pkg/metrics/producers/scheduledcapacity/producer.go

```diff
@@ -25,5 +28,56 @@ type Producer struct {
 
 // Reconcile of the metrics
 func (p *Producer) Reconcile() error {
+	var (
+		loc     *time.Location
+		err     error
+		currVal int32
+	)
+	matchFound := false
+	now := time.Now()
+
+	if p.Spec.Schedule.Timezone != nil {
+		loc, err = time.LoadLocation(*p.Spec.Schedule.Timezone)
+		if err != nil {
+			return fmt.Errorf("timezone was not a valid input")
+		}
+	} else {
+		// Set Default location to UTC if not specified
+		loc = time.UTC
+	}
+	now = now.In(loc)
+
+	for _, behavior := range p.Spec.Schedule.Behaviors {
+		// use Cron library to find the next time start and end next match
+		startTime, err := nextTime(crontabFrom(behavior.Start), loc)
+		if err != nil {
+			return fmt.Errorf("start pattern is invalid: %w", err)
+		}
+		endTime, err := nextTime(crontabFrom(behavior.End), loc)
+		if err != nil {
+			return fmt.Errorf("end pattern is invalid: %w", err)
+		}
+
+		if !now.After(endTime) && (!endTime.After(startTime) || !startTime.After(now)) {
+			// Since the way collisions are handled are by how they're ordered in the spec, stop on first match
```

Comment:
I think you might need a break statement here.

Reply:
Thanks!